libs/blocks/modal/modal.merch.js

```diff
@@ -0,0 +1,71 @@
+import { debounce } from '../../utils/action.js';
+
+export const MOBILE_MAX = 599;
+export const TABLET_MAX = 1199;
+
+export function adjustModalHeight(contentHeight) {
+  if (!window.location.hash) return;
+  const dialog = document.querySelector(window.location.hash);
+  const iframe = dialog?.querySelector('iframe');
+  const iframeWrapper = dialog?.querySelector('.milo-iframe');
+  if (!contentHeight || !iframe || !iframeWrapper) return;
+  if (contentHeight === '100%') {
+    iframe.style.height = '100%';
+    iframeWrapper.style.removeProperty('height');
+    dialog.style.removeProperty('height');
+  } else {
+    const verticalMargins = 20;
+    const clientHeight = document.documentElement.clientHeight - verticalMargins;
+    if (clientHeight <= 0) return;
+    const newHeight = contentHeight > clientHeight ? clientHeight : contentHeight;
+    iframe.style.height = '100%';
+    iframeWrapper.style.height = `${newHeight}px`;
+    dialog.style.height = `${newHeight}px`;
+  }
+}
+
+export function sendViewportDimensionsToIframe(source) {
+  const viewportWidth = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+  source.postMessage({ mobileMax: MOBILE_MAX, tabletMax: TABLET_MAX, viewportWidth }, '*');
+}
+
+export function sendViewportDimensionsOnRequest(source) {
+  sendViewportDimensionsToIframe(source);
+  window.addEventListener('resize', debounce(() => sendViewportDimensionsToIframe(source), 10));
+}
+
+function reactToMessage({ data, source }) {
+  if (data === 'viewportWidth' && source) {
+    /* If the page inside iframe comes from another domain, it won't be able to retrieve
+    the viewport dimensions, so it sends a request to receive the viewport dimensions
+    from the parent window. */
+    sendViewportDimensionsOnRequest(source);
+  }
+
+  if (data?.contentHeight) {
+    /* If the page inside iframe sends the postMessage with its content height,
+    we activate the height auto adjustment to eliminate the blank space at the bottom of the modal.
+    For this we set the iframe height to 0% in CSS to let the page inside iframe
+    to measure its content height properly.
+    Then we set the modal height to be the same as the content height we received.
+    For the modal height adjustment to work the following conditions must be met:
+    1. The modal must have the class 'commerce-frame';
+    2. The page inside iframe must send a postMessage with the contentHeight (in px, or '100%); */
+    adjustModalHeight(data?.contentHeight);
+  }
+}
+
+export function adjustStyles({ dialog, iframe }) {
+  const isAutoHeightAdjustment = /\/mini-plans\/.*mid=ft.*web=1/.test(iframe.src); // matches e.g. https://www.adobe.com/mini-plans/photoshop.html?mid=ft&web=1
+  if (isAutoHeightAdjustment) {
+    dialog.classList.add('height-fit-content');
+  } else {
+    iframe.style.height = '100%';
```

Comment:
Just curious, in what scenario could this happen?

Reply:
@Axelcureno automatic height adjustment happens in Trial with Purchase modal, for example: https://www.adobe.com/products/catalog.html#mini-plans-web-cta-creative-cloud-card